Auto Close Pending Tickets and Lime Survey added as well as various other tweaks and improvements

MarcBanyard opened the pull request:
Added various improvements that have been asked for on the forums.
New features include:-
- Copyright footer on client portal has been updated to have hyperlink to website address specified in the admin portal under "Settings > Company > Website".
- Added the code for a favicon.ico on the client and admin portals so users can put their own favicon.ico in the root folder of osTicket.
- Ability to remove staff login link fro the client login page. This has been added under "Settings > System" on the admin portal.
- Added the ability to set the agent panel width from "Settings > System" on the admin portal.
- Updated the default email and page templates to include more information that is useful as standard on new installations.
- Added the ability to easily view banned users and emails under the "User Directory" of the agent portal. (If a user or email is banned it now shows as red text under the Status column in the User Directory).
- Removed duplicate </em> tags from the /staff/staff.inc.php file.
- Added additional staff features in the admin panel for agents which were visible under the "My Preferences" section when an agent is logged in.
- Added the ability to show the Departments column on the Agent portal, also added a Ticket Status in as a Table Cell Title so it is easier to see the status of a ticket when hovering over it in the Agent Panel.
- Updated the "Support / Ticket Variables" to make it more categorised and easier to read.
- Added a new alert called "Send if ticket is Auto-Assigned via Ticket Filter or Help Topic" so agents can be notified if a ticket is automatically assigned by a Ticket Filter or Help Topic.
- Added the ability to disable Collaborators on ticket replies which can be very useful if you receive tickets from email lists and you don't want replies to go back to the whole list.
  This is great if you have agents who forget to un-tick this box when replying to a ticket.
- Updated the Ticket Filters / Ticket Data Names to make it easier to understand what is the email Subject and Body when creating Ticket Filters. This makes it much easier to understand for new users of the system.
- Added a new ticket status called "Pending" and also created a new feature called Pending Ticket Auto-Close, this will allow you to set the default status of tickets when agents reply to tickets. You can also set the amount of days tickets can sit in the new Pending queue before they get automatically closed.
- Added a new ticket auto-reply template called "Closed Ticket Auto-reply" and added the ability to enable an automatic email to be sent to end users with this new template when tickets are closed, this can be enabled or disabled per department.
- Added a new Lime Survey feature and a ticket auto-reply template called "Survey Ticket Auto-reply", this allows you to integrate Lime Survey with osTicket on a per department basis. You can enable or disable per department as well as being able to set different surveys per department.

MarcBanyard committed:
Added Company Website URL to client footer

Added the "Settings > Company > Website" URL to the client facing footer
as a hyperlink.

MarcBanyard committed:
Added code for Favourite Icon in client header

Added the code in to the client header so users can put a favicon.ico
file in the root of their osTicket installation for company branding.

MarcBanyard committed:
Ability to remove staff login link

Added the ability to remove the Staff Login Link from the client login
page from the "Settings > System" page

MarcBanyard committed:
Ability to set the agent panel page width

Added the ability to set the agent panel page width from the admin panel
in "Settings > System"
The feature is called "Default Page Width (Pixels)"

MarcBanyard committed:
Fixed ability to remove staff login link

Fixed the new feature to enable you to remove the staff login link from
the client login area.

MarcBanyard committed:
Updated the Email and Page templates

Updated the email templates that are sent to clients and staff when
tickets are received or responded to.
Also updated the page templates so that they all include the
%{company.name} to make them look fore professional

MarcBanyard committed:
Added Banned identification to "User Directory"

Added a easy to see "Banned)" in red nest to the Guest text in the
Status column within the User Directory to make it easier to see if a
user or client is banned from sending emails into the ticket system.
This has been added in the User Directory List as well as when you view
individual user details.

MarcBanyard committed:
removed duplicate </em> tags

Removed some duplicate </em> tags as tidied a few lines of HTML to fit
in with the general formatting of the page

MarcBanyard committed:
Added Staff features in admin panel

We have added the missing features to the Admin Panel when editing a
member of staff.
These features were available to the agent when they logged in under "My
Preferences" but were missing on the admin panel.
This can be very handy to see what a user has set for their preferences
as well as being able to change them.
We have also added the ability to edit the agent signature as HTML as
this was missing in the admin panel and the staff panel.
Added the ability to edit the department signature as HTML as this was
also missing.
Added the ability to edit the Authentication and Registration Templates
within the "Settings > Access" screen of the admin panel as this was
also missing.

MarcBanyard committed:
Feature to Show Ticket Dept Column

Feature to enable the Ticket Departments Column in the agents tickets
queue.
Also added the Ticket Status as a Table Cell Title so that it pops up
when you hover over the columns.

MarcBanyard committed:
Supported / Ticket Variables Updated

Updated the Ticket Variables helper to list the variables in categories
per type.

MarcBanyard committed:
Added Alert if Auto Assigned

Added the option in Alerts and Notices to alert the agents when a ticket
is automatically assigned by a Ticket Filter or Help Topic.

MarcBanyard committed:
Ability to disable Collaborators on ticket replies

Added the ability to enable or disable the checkbox for Collaborators to
be included in ticket replies by agents.
This can be very handy if your ticketing system receives mail from email
lists where you send an email to the list address and everyone in the
list gets a copy of the sent message.
Extremely helpful if you have agents who keep forgetting to untick the
box when replying to a ticket.

MarcBanyard committed:
Added help hint for the Collaborators feature

Added the hover over help hint for the Collaborators feature just added.

MarcBanyard committed:
Ticket Filters Ticket Data Names

Updates the Ticket Data Names to help users when creating Ticket
Filters.
Added the text  (Subject) and (Body) next to the label names to make it
clearer for users of the system (especially new ones).

MarcBanyard committed:
Ticket Priority Colours - Updated

Updated the Ticket Priority Colours to be a little more meaningful as
the High and Emergency were both set to red.

MarcBanyard committed:
Pending Ticket Status with Ticket Auto-Close

Added a new Ticket Status called "Pending" which is a closed ticket
status.
Created a module which will allow you to set the default agent reply
status as required, if it is set to Pending all replies will have the
status set as Pending and will show in a new queue called Pending.
You can also set the Auto Close Grace Period so that tickets that have
sat in the pending queue for X amount of days are automatically closed
by the system.
New Email Template created for "Closed Tickets"
The ability has also been added to the system to send Closed Ticket
emails to customers on a per department basis.

MarcBanyard committed:
Lime Survey Module

Added the ability to send Survey Request Emails automatically for closed
tickets.
This can be enabled or disabled per department and lets you specify
individual surveys per department.
This has been written to work with Lime Survey as it is open source like
osTicket.

MarcBanyard committed:
Updated the Signature for Surveys & Closed Tickets

Removed the "if elseif else" statement so that the department signature
gets applied as it is the system that is sending the email not an agent.

MarcBanyard committed:
Default Options for new installation

Updated the default options to work better for s UK setup as well as
enabling certain features which most users will find useful if setting
up the system for the first time.

MarcBanyard committed:
Updated the default Installed and Upgraded tickets

Added the default Status ID to both so that they show up in the tickets
list when the system is first installed or upgraded.

infectormp commented:
@protich, @greezybacon please merge this PR to main branch!

protich commented:
@infectormp - I fortunately we have to do a review before merging. 

@datatecuk - To speed up review process, the pull request should have been broken down to individual requests dealing with specific  issue or feature.

datatecuk commented:
@protich Sorry, I did start to do Pull Requests individually, but GitHub will only let you do 1 Pull Request at a time.
I can cancel this pull request and submit individual ones if it makes things easier at your end.
All of the changes made were done over a week or so and have been tested on a development installation and all work fine. We have already pushed our changes to our production environment and all is working very well.
There are some other tings we are looking to fix on osTicket to clean he database up when tickets are deleted as it leaves various database entries in the "ticket_event" and "ticket_email_info".
We can also see an issue with lots of entries being left in the "draft" table as well.
Let me know if you want to talk through this (please let me know if you are happy for me to contact you privately to run through the above.
We are also keen to port the above commits to the v1.10 version of osTicket so that they are available in the new version when it is released.

protich commented:
I'll most definitely recommend making individual pull requests. A pull request should only have related commits -- addressing a specific issue or adding a specific feature (we follow GitFlow model). It's ok for PR to be dependent on others -- i.e branch from a branch. 

Making separate request helps with the review process and __more importantly__ helps narrow the subsequent discussion to a specific issue or feature. For example, already see a couple of commits / changes I disagree with that we can open discussion on once resubmitted.

ticket_event and ticket_email_info tables are not cleaned on purpose. Ticket event provides historic stats used on the dashboard and the later is used to track and prevent duplicate email tickets (e.g deleting a ticket that's still in the inbox would result in refetching).

You can reach me at peter (at) osticket.com

philbertphotos commented:
I am real interested in the survey part :)

ntozier commented:
I feel like the Lime survey integration part should be a separate pull request by itself.

philbertphotos commented:
I agree..

Sent from my iPhone

> On Feb 29, 2016, at 9:28 AM, ntozier <notifications@github.com> wrote:
> 
> I feel like the Lime survey integration part should be a separate pull request by itself.
> 
> —
> Reply to this email directly or view it on GitHub.
>

datatecuk commented:
No problems guys, I will cancel this pull request an post the modifications as new PR's, but it may be some time as GitHub will only let you do 1 Pull Request at a time, does anyone know if you can create another Pull Request once you have already submitted one that has not been merged down to a project (I know we have done multiple PR's before, but when I tried last week it would not let me as there was an open PR which I had submitted on the osTicket project.

protich commented:
@datatecuk - you simply need to create a branch for each of the pull request.  Below is a model I would follow if I had to make a pull request on issue "abc"

-> Create a branch from develop-next called  "issue/abc" (if feature use feature/abc) 
-> Chery-pick commits you made for "abc"
-> Make pull request 

If a feature "abc" is dependent on "abc" then simply brach from "abc" branch.

jdelhome3578 commented:
Auto close is already PR #2152 for develop and #2917 for develop-next

infectormp commented:
@jdelhome3578 i use PR #2152 but this realization is more powerfull

philbertphotos commented:
Thank you .... keep them coming :)

delcain commented:
How can i use  Lime Survey feature , 

I tray many branchs but i not found "Survey Ticket Auto-reply"

JediKev commented:
@delcain

This is not a part of core rather a pull request. You will need to fetch the pull request locally and apply to your production, apply the code as a diff/patch, or apply the changes manually.

Unfortunately it looks like this has many conflicts that you will have to sort out yourself when applying to production.

Cheers.

MarcBanyard commented:
It was a pull request, but we were asked to put each bit in as separate pull requests and simply didn’t have the time.

As far as I can remember we did each item as a separate commit to git and then pushed before doing a combined merge request.


Kind Regards,
Marc



Marc Banyard
Technical Director
Datatech UK Ltd




Phone:          +44 (0) 1200 400 600
Fax:    +44 (0) 1200 400 611
Mobile:         +44 (0) 7799 114 775
Email:          Marc.Banyard@datatechuk.com<mailto:Marc.Banyard@datatechuk.com>
Website:        www.datatechuk.com<http://www.datatechuk.com>

[Datatech UK Ltd]        Prospect House
 Fishing Line Road
 Redditch,
 Worcestershire
 B97 6EW
 United Kingdom

Data Centres | CoLocation | Rack Space | BackUp Solutions | IT Solutions | Telecoms | VoIP | CCTV | Leased Lines | MPLS | DSL | SIP


[Datatech UK Ltd - ISO 9001]      [Dell Premier Partner]       [Cisco Select Certified Partner]       [Datatech UK Ltd - ISO 27001]


 [cid:image323e4b.GIF@19235398.4f896c46]   Please consider the environment before printing this email message.


IMPORTANT:
This message is private and confidential. If you have received this message in error, please notify us and remove it from your system.
Please note that Datatech UK Ltd may monitor email traffic data and also the content of email for the purposes of security and staff training.
All calls made from or made to Datatech UK Ltd may be recorded for training and quality assurance purposes.
Any prices quotes are valid for a maximum of 5 days from {Current Date} unless otherwise stated. All prices quoted are exclusive of VAT. E&OE.


Disclaimer:
This message contains confidential information and is intended only for {Message Recipients}. If you are not {Message Recipients} you should not disseminate, distribute or copy this e-mail. Please notify {Message Sender} immediately by e-mail if you have received this e-mail by mistake and delete this e-mail from your system. E-mail transmission cannot be guaranteed to be secure or error-free as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. {Display name} therefore does not accept liability for any errors or omissions in the contents of this message, which arise as a result of e-mail transmission. If verification is required please request a hard-copy version.


---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Datatech UK Ltd is a limited company registered in England and Wales. Registered number: 05696509. VAT Registration Number 878 4525 74.
Registered office: Datatech UK Ltd, Prospect House, Fishing Line Road, Redditch, Worcestershire, B97 6EW, United Kingdom
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

On 14 Nov 2019, at 21:46, JediKev <notifications@github.com> wrote:

﻿

@delcain<https://github.com/delcain>

This is not a part of core rather a pull request. You will need to fetch the pull request locally and apply to your production, apply the code as a diff/patch, or apply the changes manually.

Unfortunately it looks like this has many conflicts that you will have to sort out yourself when applying to production.

Cheers.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/osTicket/osTicket/pull/2942?email_source=notifications&email_token=ABKYC3XXFMWM4W6QLY55PNTQTXBLRA5CNFSM4B4VBFZ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEDNCFY#issuecomment-554094871>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABKYC3QES7EZ7LQ2NU72PCDQTXBLRANCNFSM4B4VBFZQ>.

MarcBanyard committed:
Revert "Updated the default Installed and Upgraded tickets"

This reverts commit 5a56daf683ebaa7017cec2b31357fd3c5927ff6c.

Revert "Default Options for new installation"

This reverts commit 362c721d4272886e5abb90bd25d1ff0eeaf887b7.

Revert "Updated the Signature for Surveys & Closed Tickets"

This reverts commit 20cfc5c3b718693d5cefe28ca38876bb71cb663d.

Revert "Lime Survey Module"

This reverts commit 2c403c31e17d79e4c7d3f0028c6a0860f95aae4f.

Revert "Pending Ticket Status with Ticket Auto-Close"

This reverts commit 324e0c319dc195bc16a12cc040fb9215673f4ba2.

Revert "Ticket Priority Colours - Updated"

This reverts commit 4d33705cf58336e8b795ffd1bbe6f486a8c04c2e.

Revert "Ticket Filters Ticket Data Names"

This reverts commit 76d99762439a1ad507e069806539ef9128fe6ada.

Revert "Added help hint for the Collaborators feature"

This reverts commit d1cedd2e07266324ec6b5f841f4702554d27442a.

Revert "Ability to disable Collaborators on ticket replies"

This reverts commit ceeef34c39a97e32685eae66065954dba75d2147.

Revert "Added Alert if Auto Assigned"

This reverts commit cefa883fea8cd1f3211435e8e34683327ffb58b5.

Revert "Supported / Ticket Variables Updated"

This reverts commit 8fcb38167ee968d89cfcd26dee08536623a41ecc.

Revert "Feature to Show Ticket Dept Column"

This reverts commit 284745ffda555fc6c83f6d72ee08c69a5cbcb51e.

Revert "Added Staff features in admin panel"

This reverts commit c2e8a73d908922b55c8ac4f3b2a94d69111ac539.

Revert "removed duplicate </em> tags"

This reverts commit 280c25da190729de1f5933be223981d616ff1012.

Revert "Added Banned identification to "User Directory""

This reverts commit 00e4d215907e77dc0ee64f296cef90f8fddd1f4a.

Revert "Updated the Email and Page templates"

This reverts commit 7daa2f67ff1089773d4135e72ac6ad3a70513324.

Revert "Fixed ability to remove staff login link"

This reverts commit 2ca748030664cb1b2135e9e354db9db99032b1fd.

Revert "Ability to set the agent panel page width"

This reverts commit 2faf499603c83d3fcfa2ea2be53252f6251c4298.

Revert "Ability to remove staff login link"

This reverts commit 6bfb905b3ee1c9ee412f60e28a129311fe0b1f56.

Revert "Added code for Favourite Icon in client header"

This reverts commit a5dd1d0ea51909a45d5beb16bc6f26bc17812e5d.

Revert "Added Company Website URL to client footer"

This reverts commit 8351f66149298ed42044c37406018bcba6f021a3.

MarcBanyard committed:
Merge branch 'osTicket:develop' into develop

MarcBanyard committed:
Added code for Favourite Icon in client header
Added the code in to the client header so users can put a favicon.ico
file in the root of their osTicket installation for company branding.

MarcBanyard committed:
Ability to remove staff login link
Added the ability to remove the Staff Login Link from the client login
page from the "Settings > System" page

MarcBanyard committed:
Ability to set the agent panel page width
Added the ability to set the agent panel page width from the admin panel
in "Settings > System"
The feature is called "Default Page Width"

MarcBanyard committed:
Added Banned identification to "User Directory"
Added a easy to see "Banned)" in red nest to the Guest text in the Status column within the User Directory to make it easier to see if a user or client is banned from sending emails into the ticket system.
This has been added in the User Directory List as well as when you view individual user details.

MarcBanyard committed:
Ticket Priority Colours - Updated
Updated the Ticket Priority Colours to be a little more meaningful as the High and Emergency were both set to red.

MarcBanyard committed:
Added Staff features in admin panel
We have added the missing features to the Admin Panel when editing a member of staff.
These features were available to the agent when they logged in under "Profile" but were missing on the admin panel.
This can be very handy to see what a user has set for their preferences as well as being able to change them.
We have also added the ability to edit the agent signature as HTML as this was missing in the admin panel and the staff panel.
Added the ability to edit the department signature as HTML as this was also missing.
Added the ability to edit the Authentication and Registration Templates & Pages within the "Settings > Agents > Templates" screen of the admin panel as this was also missing.

MarcBanyard committed:
Added Alert if Auto Assigned
Added the option in "Settings > Tickets > Alerts and Notices" to alert the agents when a ticket is automatically assigned by a Ticket Filter or Help Topic.

MarcBanyard committed:
Auto-Close Answered Tickets

You can set the Auto Close Grace Period so that tickets that have been answered for X amount of days are automatically closed by the system.
New Email Template created for "Closed Tickets"
The ability has also been added to the system to send Closed Ticket emails to customers on a per department basis.

MarcBanyard committed:
Auto-Close Answered Tickets

Fix enter 0 of the Auto Close Grace Period is null or empty

MarcBanyard committed:
Make reply panel on tickets full width as it was hard coded to 940px for some reason.
